aws/adapter.go

```diff
@@ -948,9 +948,12 @@ func (a *Adapter) FindLBSubnets(scheme string) []string {
 
 		// If we have two subnets for the same AZ we arbitrarily choose
 		// the one that is first lexicographically.
-		if strings.Compare(existing.id, subnet.id) > 0 {
+		if subnet.id < existing.id {
 			subnetsByAZ[subnet.availabilityZone] = subnet
 		}
+
+		log.Warnf("Found subnets %s and %s for availability zone %s, using %s",
+			existing.id, subnet.id, subnet.availabilityZone, subnetsByAZ[subnet.availabilityZone])
```

Comment:
This is not precise as there could be more then two sunbets with the tag and the final subnet is only known after loop ends

Reply:
Fixed